### ISSUE TITLE
`wait-for-db` in initContainer prevents the use of serviceMeshes

### ISSUE THREAD

**Issue by jrossBah** [grafana/oncall#3855]
### What would you like to see!

tl;dr
move `wait-for-db` to the container's command rather than an init container. So that it works with serviceMesh's like istio that will prevent an initContainer from communicating as the sidecar for it does not exist

Right now I am having to do heavy patching because of the way kubernetes initContainers work in sequence with use with Istio. 

The commands in the initContainer have no need to be there. And in fact make the operation of OnCall much more difficult. Move the initContainer commands to run in the container themselves.

For the Celery deployment this is pretty easy. 
This may be less so for the Engine deployment as currently the startup command is housed in the image itself rather than the deployment. This is probably also something that should be changed or at least explained.

### Product Area

Alert Flow & Configuration, Helm

**Comment by bpedersen2** [comment]
See https://github.com/istio/istio/issues/23802 for possible other  workarounds (especially running the init container as uid 1337)

**Comment by bpedersen2** [comment]
And a reason for why the check is done  in an init container: it does not only wait for the db to be reachable, it waits until the db migration job kicked off during helm upgrades is finished and the db schema matches the expected version. As this may take significant time, running this in the main container would require long health check timeouts which you probably won't want as well.

**Comment by joeyorlando** [comment]
We don't have any near term plans to remove the initContainer. @bpedersen2 summarized well the reason why we have/need this (thank you 🙏).

With that said, we're open to reviewing a PR to modify the helm chart to accomplish what you're after. Feel free to open one 🙂
